internal/command/launch/cmd.go

```diff
@@ -120,7 +121,9 @@ func getManifestArgument(ctx context.Context) (*LaunchManifest, error) {
 
 func run(ctx context.Context) (err error) {
 
-	if !flag.GetBool(ctx, "ui") {
+	// NOTE: We depend on legacy launcher behavior for Nomad support, which is needed for the MigrateToV2 tests
+	//       Once we rip out those tests, this can go with them.
+	if flag.GetBool(ctx, "no-ui") || flag.GetBool(ctx, "force-nomad") {
```

Comment:
Can it resort to legacy "fly launch" if `--reuse-app` is passed and print a big deprecation warning if used?